drivers/ethernet/phy/phy_ti_dp83867.c

```diff
@@ -0,0 +1,576 @@
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct ti_dp83867_data *data =
+		CONTAINER_OF(dwork, struct ti_dp83867_data, phy_monitor_work);
+	const struct device *dev = data->dev;
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	const struct ti_dp83867_config *config = dev->config;
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios) */
+	struct phy_link_state state = {};
+	int ret;
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	if (config->interrupt_gpio.port) {
+		ret = phy_ti_dp83867_clear_interrupt(data);
+		if (ret) {
+			return;
+		}
+	}
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios) */
+
+	ret = phy_ti_dp83867_get_link(dev, &state);
+
+	if (ret == 0 && memcmp(&state, &data->state, sizeof(struct phy_link_state)) != 0) {
+		memcpy(&data->state, &state, sizeof(struct phy_link_state));
+		if (data->cb) {
+			data->cb(dev, &data->state, data->cb_data);
+		}
+	}
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	if (!config->interrupt_gpio.port) {
+		k_work_reschedule(&data->phy_monitor_work, K_MSEC(CONFIG_PHY_MONITOR_PERIOD));
+	}
+#else
+	k_work_reschedule(&data->phy_monitor_work, K_MSEC(CONFIG_PHY_MONITOR_PERIOD));
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios) */
+}
+
+static int phy_ti_dp83867_init(const struct device *dev)
+{
+	const struct ti_dp83867_config *config = dev->config;
+	struct ti_dp83867_data *data = dev->data;
+	int ret;
+
+	data->dev = dev;
+
+	ret = k_mutex_init(&data->mutex);
+	if (ret) {
+		return ret;
+	}
+
+	mdio_bus_enable(config->mdio_dev);
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	if (config->reset_gpio.port) {
+		ret = gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_INACTIVE);
+		if (ret) {
+			return ret;
+		}
+	}
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios) */
+
+	/* Reset PHY */
+	ret = phy_ti_dp83867_reset(dev);
+	if (ret) {
+		LOG_ERR("Failed to reset phy (%d)", config->addr);
+		return ret;
+	}
+
+	k_work_init_delayable(&data->phy_monitor_work, phy_ti_dp83867_monitor_work_handler);
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	if (!config->interrupt_gpio.port) {
+		phy_ti_dp83867_monitor_work_handler(&data->phy_monitor_work.work);
+		goto skip_int_gpio;
+	}
+
+	/* Configure interrupt pin */
+	ret = gpio_pin_configure_dt(&config->interrupt_gpio, GPIO_INPUT);
+	if (ret) {
+		return ret;
+	}
+
+	gpio_init_callback(&data->gpio_callback, phy_ti_dp83867_interrupt_handler,
+			   BIT(config->interrupt_gpio.pin));
+	ret = gpio_add_callback_dt(&config->interrupt_gpio, &data->gpio_callback);
+	if (ret) {
+		return ret;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&config->interrupt_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret) {
+		return ret;
+	}
+
+skip_int_gpio:
+#else
+	phy_ti_dp83867_monitor_work_handler(&data->phy_monitor_work.work);
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios) */
+
+	return 0;
+}
+
+static const struct ethphy_driver_api ti_dp83867_phy_api = {
```

Comment:
Wrap device driver using the device API macros

```suggestion
static DEVICE_API(ethphy, ti_dp83867_phy_api) = {
```

Reply:
Fixed and added an entry in a seperate commit.